libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -866,6 +866,10 @@ void AP_GPS::update_instance(uint8_t instance)
     // we have an active driver for this instance
     bool result = drivers[instance]->read();
     uint32_t tnow = AP_HAL::millis();
+    
+    if ((state[instance].num_sats < 10) && (state[instance].hdop > 120)) {
```

Comment:
This doesn't look right.  If you want to add quality measures like this then they belong elsewhere in the code (and they *do* exist elsewhere in the code).

Modifying the fix type just because you don't like some aspect of the fix isn't correct IMO, and will lead to unexpected behavioural changes where e.g. we use the time from the satellites even when we're not using the co-ordinates.